internal/controllers/machineset/machineset_controller.go

```diff
@@ -880,8 +885,7 @@ func (r *Reconciler) updateStatus(ctx context.Context, cluster *clusterv1.Cluste
 
 		node, err := r.getMachineNode(ctx, cluster, machine)
 		if err != nil && machine.GetDeletionTimestamp().IsZero() {
-			log.Error(err, "Unable to retrieve Node status", "node", klog.KObj(node))
-			continue
+			return errors.Wrapf(err, "unable to retrieve the status of Node %s", klog.KObj(node))
```

Comment:
Not sure what impact that has on other edge cases. E.g. what if we get a NotFound error?

Maybe it's safer to only return here if we hit the ErrClusterLocked error?

Reply:
Good point. I just updated getMachineNode() to return nil Node and nil error when it hits a NotFound error, so it goes to line 891 `if noderefutil.IsNodeReady(node)` which returns false. For other errors, I think it's OK to let ti return error and requeue.